### PR TITLE
Google ads etl

### PR DESCRIPTION
**Pull request by fbertsch**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3461)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/google_ads_derived/ad_groups_v1/query.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+WITH ad_group_names AS (
+  SELECT
+    campaign_id,
+    id AS ad_group_id,
+    name AS ad_group_name,
+    ROW_NUMBER() OVER (PARTITION BY id, campaign_id ORDER BY updated_at DESC) AS rn,
+  FROM
+    `moz-fx-data-bq-fivetran`.ads_google_mmc.ad_group_history
+  QUALIFY
+    rn = 1
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/google_ads_derived/ad_groups_v1/query.sql:10]:
```suggestion
    ROW_NUMBER() OVER (PARTITION BY id, campaign_id ORDER BY updated_at DESC) = 1
```

File: sql/moz-fx-data-shared-prod/google_ads_derived/ad_groups_v1/query.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+WITH ad_group_names AS (
+  SELECT
+    campaign_id,
+    id AS ad_group_id,
+    name AS ad_group_name,
+    ROW_NUMBER() OVER (PARTITION BY id, campaign_id ORDER BY updated_at DESC) AS rn,
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/google_ads_derived/ad_groups_v1/query.sql:6]:
```suggestion
```

File: sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/query.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
+WITH campaign_names AS (
+  SELECT
+    customer_id,
+    id AS campaign_id,
+    name AS campaign_name,
+    ROW_NUMBER() OVER (PARTITION BY id ORDER BY updated_at DESC) = 1 AS is_most_recent_record,
+  FROM
+    `moz-fx-data-bq-fivetran`.ads_google_mmc.campaign_history
+  QUALIFY
+    is_most_recent_record
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/query.sql:10]:
```suggestion
    ROW_NUMBER() OVER (PARTITION BY id ORDER BY updated_at DESC) = 1
```

File: sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/query.sql [hunk 4]
```diff
@@ -0,0 +1,37 @@
+WITH campaign_names AS (
+  SELECT
+    customer_id,
+    id AS campaign_id,
+    name AS campaign_name,
+    ROW_NUMBER() OVER (PARTITION BY id ORDER BY updated_at DESC) = 1 AS is_most_recent_record,
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/query.sql:6]:
```suggestion
```

File: sql/moz-fx-data-shared-prod/google_ads_derived/ad_groups_v1/query.sql (hunk 1 above)
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/google_ads_derived/ad_groups_v1/query.sql:10]: I'm actually going to switch this to what I did for `campaigns_v1` and call it `is_most_recent_record`

File: sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/query.sql [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    -- Mozilla Firefox UAC account extracts some info from campaign names
+    account_id = 2474006105,
+    STRUCT(
+      mozfun.google_ads.extract_region_from_campaign_name(campaign_name) AS campaign_region,
+      mozfun.google_ads.extract_country_code_from_campaign_name(
+        campaign_name
+      ) AS campaign_country_code,
+      mozfun.google_ads.extract_language_from_campaign_name(campaign_name) AS campaign_language
+    ),
+    STRUCT(
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/query.sql:27]: So we expect some campaigns not to localization info?

File: sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/schema.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+fields:
+- mode: NULLABLE
+  name: account_name
+  type: STRING
+  description: The name of the Google Ads account, e.g. Mozilla Firefox.
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/google_ads_derived/campaigns_v1/schema.yaml:5]: Thanks for adding descriptions! 🙇

**Review by kwindau — APPROVED** [review]

File: sql/mozfun/google_ads/extract_country_code_from_campaign_name/udf.sql [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    "AU"
+  ),
+  assert.equals(
+    google_ads.extract_country_code_from_campaign_name("Mozilla_Firefox_DE_DE_Google_UAC_Android"),
+    "DE"
+  ),
+  assert.equals(
+    google_ads.extract_country_code_from_campaign_name("Mozilla_NA_FF_UAC_Group1_Event1"),
+    CAST(NULL AS STRING)
+  ),
```
Comment (kik-kik) [inline sql/mozfun/google_ads/extract_country_code_from_campaign_name/udf.sql:38]: Should we add a test case for when null is passed? Are we sure campaign_name will either be a text value or null or could it be an empty string?

**Review by kik-kik — APPROVED** [review]
r+wc

File: sql/mozfun/google_ads/extract_country_code_from_campaign_name/udf.sql (hunk 7 above)
Comment (fbertsch) [reply sql/mozfun/google_ads/extract_country_code_from_campaign_name/udf.sql:38]: Good point, I can add both.

I'm actually thinking I should consolidate these 3 UDFs into a single UDF that returns a struct.
